src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -4,12 +4,24 @@
 using System;
 using System.Diagnostics;
 using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Text;
 using System.Threading;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
 {
     public struct MemoryPoolIterator
     {
+        private const ulong _xorPowerOfTwoToHighByte = (0x07ul       |
+                                                        0x06ul <<  8 |
+                                                        0x05ul << 16 |
+                                                        0x04ul << 24 |
+                                                        0x03ul << 32 |
+                                                        0x02ul << 40 |
+                                                        0x01ul << 48 ) + 1;
+
+        private static readonly Encoding _utf8 = Encoding.UTF8;
```

Comment:
Used by some methods that moved from Extensions to member would probably be better in Constants